test/system/250-systemd.bats

```diff
@@ -136,6 +136,22 @@ function service_cleanup() {
     service_cleanup
 }
 
+# Regression test for #11438
+@test "podman generate systemd - restart policy" {
+    cname=$(random_string)
+    run_podman create --restart=always --name $cname $IMAGE
+    run_podman generate systemd --new $cname
+    is "$output" ".*Restart=always.*" "Use container's restart policy if set"
+    run_podman generate systemd --new --restart-policy=on-failure $cname
+    is "$output" ".*Restart=on-failure.*" "Override container's restart policy"
```

Comment:
Can you also check that the `--restart` argument is removed from the podman run command. This should also check `--restart always` not only `--restart=always`